xarray/core/dataset.py

```diff
@@ -2366,6 +2339,58 @@ def unstack(self, dim):
 
         return self._replace_vars_and_dims(variables, coord_names)
 
+    def unstack(self, dim=None):
+        """
+        Unstack existing dimensions corresponding to MultiIndexes into
+        multiple new dimensions.
+
+        New dimensions will be added at the end.
+
+        Parameters
+        ----------
+        dim : str or sequence of str, optional
+            Dimension(s) over which to unstack. By default unstacks all
+            MultiIndexes.
+
+        Returns
+        -------
+        unstacked : Dataset
+            Dataset with unstacked data.
+
+        See also
+        --------
+        Dataset.stack
+        """
+        dim_from_kwarg = dim is not None
+
+        if isinstance(dim, basestring):
+            dims = set([dim])
+        elif dim is None:
+            dims = set(self.dims)
+        else:
+            dims = set(dim)
+
+        missing_dims = [dim for dim in dims if dim not in self.dims]
```

Comment:
Minor point: use a new variable name other than `dim` here (even `d` would be fine) to avoid masking the original `dim` argument.